job-service-db/src/main/resources/procedures/createJobForDependencies.sql

```diff
@@ -127,14 +130,18 @@ BEGIN
         -- this job is blocked from starting. Shared locks are being used as the rows themselves will not be updated by the operation.
         -- This means that there will be no contention when simultaneously creating multiple jobs which have the same pre-requisites.
         SELECT * FROM job
-        WHERE job_id IN (SELECT job_id FROM prereqs) AND status <> 'Completed'
-        ORDER BY job_id
+        WHERE partition = in_partition
+            AND job_id IN (SELECT job_id FROM prereqs)
+            AND status <> 'Completed'
+        ORDER BY partition, job_id
```

Comment:
Is there an advantage to adding `partition` to the `ORDER BY` clause (given that it is already locked down)?  Does this help it choose the correct index for ordering?

Reply:
I'm not sure if it would fail to find the index otherwise, but writing the columns out in index-order makes it more obvious (to me, at least) that it will definitely use the index.